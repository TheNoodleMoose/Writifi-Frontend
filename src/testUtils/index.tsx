import React, { ReactElement, ReactNode } from "react";

import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export const renderWithContext = (component: ReactElement) => {
  const history = createMemoryHistory();
  const wrapper = ({ children }: { children?: ReactNode }) => (
    <Router history={history}>{children}</Router>
  );

  const renderResults = render(component, { wrapper });
  return {
    ...renderResults,
    snapshot: () => {
      expect(renderResults.container).toMatchSnapshot();
    }
  };
};
