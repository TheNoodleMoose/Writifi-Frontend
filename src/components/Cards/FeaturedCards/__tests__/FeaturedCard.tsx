import React from "react";
import { renderWithContext } from "../../../../testUtils";

import FeaturedCard, { FeaturedCardType } from "..";

it("renders correctly | Featured", () => {
  renderWithContext(
    <FeaturedCard type={FeaturedCardType.featured} />
  ).snapshot();
});

it("renders correctly | Editors Choice", () => {
  renderWithContext(
    <FeaturedCard type={FeaturedCardType.editorsChoice} />
  ).snapshot();
});
