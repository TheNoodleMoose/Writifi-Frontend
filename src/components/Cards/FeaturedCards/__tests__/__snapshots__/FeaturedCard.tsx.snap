// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly | Editors Choice 1`] = `
<div>
  <div
    class="sc-AxjAm ciKidB"
  >
    <div
      class="sc-AxirZ kbkaQd"
      type="EditorsChoice"
    >
      <h2>
        Editor's Choice
      </h2>
    </div>
    <div
      class="sc-AxiKw llAaWr"
    >
      <p
        class="sc-AxhCb cwwnQe"
        type="EditorsChoice"
      >
        Lorem ipsum dolor sit Molestias?
      </p>
      <p
        class="sc-AxhCb cwwnQe"
        type="EditorsChoice"
      >
        Lorem ipsum dolor sit Molestias?
      </p>
      <p
        class="sc-AxhCb cwwnQe"
        type="EditorsChoice"
      >
        Lorem ipsum dolor sit Molestias?
      </p>
      <p
        class="sc-AxhCb cwwnQe"
        type="EditorsChoice"
      >
        Lorem ipsum dolor sit Molestias?
      </p>
    </div>
  </div>
</div>
`;

exports[`renders correctly | Featured 1`] = `
<div>
  <div
    class="sc-AxjAm ciKidB"
  >
    <div
      class="sc-AxirZ fxlTtO"
      type="Featured"
    >
      <h2>
        Featured Reading
      </h2>
    </div>
    <div
      class="sc-AxiKw llAaWr"
    >
      <p
        class="sc-AxhCb cwwnQe"
        type="Featured"
      >
        Lorem ipsum dolor sit Molestias?
      </p>
      <p
        class="sc-AxhCb cwwnQe"
        type="Featured"
      >
        Lorem ipsum dolor sit Molestias?
      </p>
      <p
        class="sc-AxhCb cwwnQe"
        type="Featured"
      >
        Lorem ipsum dolor sit Molestias?
      </p>
      <p
        class="sc-AxhCb cwwnQe"
        type="Featured"
      >
        Lorem ipsum dolor sit Molestias?
      </p>
    </div>
  </div>
</div>
`;
